projectgenerator/tests/test_projectgen_generic_databases.py

```diff
@@ -0,0 +1,185 @@
+        cur = conn.cursor(cursor_factory=psycopg2.extras.DictCursor)
+
+        cur.execute("CREATE SCHEMA IF NOT EXISTS no_interlis_schema_spatial;")
+        cur.execute("""
+            CREATE TABLE no_interlis_schema_spatial.point (
+                id serial NOT NULL,
+                name text,
+                geometry geometry(POINT, 4326) NOT NULL,
+                CONSTRAINT point_id_pkey PRIMARY KEY (id)
+            );
+            CREATE TABLE no_interlis_schema_spatial.region (
+                id serial NOT NULL,
+                name text,
+                geometry geometry(POLYGON, 4326) NOT NULL,
+                id_point integer,
+                CONSTRAINT region_id_pkey PRIMARY KEY (id)
+            );
+        """)
+        cur.execute("""
+            ALTER TABLE no_interlis_schema_spatial.region ADD CONSTRAINT region_point_id_point_fk FOREIGN KEY (id_point)
+            REFERENCES no_interlis_schema_spatial.point (id) MATCH FULL
+            ON DELETE SET NULL ON UPDATE CASCADE;
+        """)
+        conn.commit()
+
+        generator = Generator('ili2pg', uri, 'smart1', 'no_interlis_schema_spatial')
+        layers = generator.layers()
+
+        self.assertEqual(len(layers), 2)
+        self.assertEqual(len(generator.relations(layers)), 1)
+
+        for layer in layers:
+            self.assertIsNotNone(layer.geometry_column)
+
+        cur.execute("DROP SCHEMA no_interlis_schema_spatial CASCADE;")
+        conn.commit()
+        cur.close()
+
+    def test_postgis_db_with_non_empty_and_no_interlis_schema_with_non_spatial_tables(self):
+        uri = 'dbname=gis user=docker password=docker host=postgres'
+        conn = psycopg2.connect(uri)
+        cur = conn.cursor(cursor_factory=psycopg2.extras.DictCursor)
+
+        cur.execute("CREATE SCHEMA IF NOT EXISTS no_interlis_schema;")
+        cur.execute("""
+            CREATE TABLE no_interlis_schema.point (
+                id serial NOT NULL,
+                name text,
+                CONSTRAINT point_id_pkey PRIMARY KEY (id)
+            );
+            CREATE TABLE no_interlis_schema.region (
+                id serial NOT NULL,
+                name text,
+                id_point integer,
+                CONSTRAINT region_id_pkey PRIMARY KEY (id)
+            );
+        """)
+        cur.execute("""
+            ALTER TABLE no_interlis_schema.region ADD CONSTRAINT region_point_id_point_fk FOREIGN KEY (id_point)
+            REFERENCES no_interlis_schema.point (id) MATCH FULL
+            ON DELETE SET NULL ON UPDATE CASCADE;
+        """)
+        conn.commit()
+
+        generator = Generator('ili2pg', uri, 'smart1', 'no_interlis_schema')
+        layers = generator.layers()
+
+        self.assertEqual(len(layers), 2)
+        self.assertEqual(len(generator.relations(layers)), 1)
+
+        for layer in layers:
+            self.assertIsNone(layer.geometry_column)
+
+        cur.execute("DROP SCHEMA no_interlis_schema CASCADE;")
```

Comment:
Better done in a `try: finally:` block.

I don't mind merging this as is for now, so this comment is an optional improvement proposal (and something to keep in mind for the future).

Reply:
agree

Reply:
> Especially because the next tests are executed indifferently of unit test result!!

I think they are (but if you want to be sure better look it up ;) )

Reply:
No. The tests are working. Need a enchacement. But it could go in a newer PR like a refactor.

Reply:
it's ready https://github.com/opengisch/projectgenerator/pull/157/commits/3003a94f160a8139f1b3f2f9499c5d46b5a16a9c